pkg/sdn/plugin/proxy.go

```diff
@@ -77,8 +80,13 @@ func (proxy *ovsProxyPlugin) watchEgressNetworkPolicies() {
 		if eventType == watch.Deleted {
 			policy.Spec.Egress = nil
 		}
+
+		proxy.lock.Lock()
 		proxy.updateNetworkPolicy(*policy)
-		// FIXME: poke the endpoint-syncer somehow...
+		if proxy.allEndpoints != nil {
+			proxy.updateEndpoints()
+		}
+		proxy.lock.Unlock()
```

Comment:
in case of a panic, `Unlock` will never get called, `utilwait.Forever` will recover the panic and restart `watchEgressNetworkPolicies`, but the `proxy.lock.Lock()` call will hang.

wrapping in a func and deferring the Unlock will make sure we unlock
```
func(){
  proxy.lock.Lock()
  defer proxy.lock.Unlock()
  if proxy.allEndpoints != nil {
    proxy.updateEndpoints()
  }
}()
```